tests/level3_test.go

```diff
@@ -0,0 +1,195 @@
+			actual++
+		}
+	}
+
+	expected := 2
+
+	assert.Equal(t, expected, actual, fmt.Sprintf("There must be %d resource group with 'landingzone=networking_hub' and 'environment=%s' tags, found %d", expected, test.Environment, actual))
+}
+
+func TestVirtualNetworksAreInDifferentRegions(t *testing.T) {
+	t.Parallel()
+
+	test := prepareTestTable()
+
+	vnet1, _ := azure.GetVirtualNetworkE(fmt.Sprintf("%s-vnet-hub-re1", test.Prefix), fmt.Sprintf("%s-rg-vnet-hub-re1", test.Prefix), test.SubscriptionID)
+
+	vnet2, _ := azure.GetVirtualNetworkE(fmt.Sprintf("%s-vnet-hub-re2", test.Prefix), fmt.Sprintf("%s-rg-vnet-hub-re2", test.Prefix), test.SubscriptionID)
+
+	assert.NotEqual(t, *vnet1.Location, *vnet2.Location, "Virtual Networks in the 'landingzone=networking_hub' resource groups should provisioned in different regions")
+}
+
+func TestBastionSubNetSecurityRulesCount(t *testing.T) {
+	t.Parallel()
+
+	test := prepareTestTable()
+
+	for iLoop := 1; iLoop <= 2; iLoop++ {
+		rules := azure.GetAllNSGRules(t, fmt.Sprintf("%s-rg-vnet-hub-re%d", test.Prefix, iLoop), fmt.Sprintf("%s-nsg-AzureBastionSubnet", test.Prefix), test.SubscriptionID)
+
+		assert.Equal(t, 12, len(rules.SummarizedRules), fmt.Sprintf("Bastion Subnet should have 12 rules, found %d", len(rules.SummarizedRules)))
+	}
+}
+
+func TestBastionSubNetInboundSecurityRules(t *testing.T) {
+	t.Parallel()
+
+	test := prepareTestTable()
+
+	for iLoop := 1; iLoop <= 2; iLoop++ {
+		rules := azure.GetAllNSGRules(t, fmt.Sprintf("%s-rg-vnet-hub-re%d", test.Prefix, iLoop), fmt.Sprintf("%s-nsg-AzureBastionSubnet", test.Prefix), test.SubscriptionID)
+		actual := make([]string, 0)
+
+		for _, rule := range rules.SummarizedRules {
+			if rule.Direction == "Inbound" {
+				actual = append(actual, rule.DestinationPortRange)
+			}
+		}
+
+		assert.ElementsMatch(t, test.Config.BastionInboundRules, actual, fmt.Sprintf("Bastion Subnet doesn't have expected destination ports: %+q", test.Config.BastionInboundRules))
+	}
+}
+
+func TestBastionSubNetOutboundSecurityRules(t *testing.T) {
+	t.Parallel()
+
+	test := prepareTestTable()
+
+	for iLoop := 1; iLoop <= 2; iLoop++ {
+		rules := azure.GetAllNSGRules(t, fmt.Sprintf("%s-rg-vnet-hub-re%d", test.Prefix, iLoop), fmt.Sprintf("%s-nsg-AzureBastionSubnet", test.Prefix), test.SubscriptionID)
+		actual := make([]string, 0)
+
+		for _, rule := range rules.SummarizedRules {
+			if rule.Direction == "Outbound" {
+				actual = append(actual, rule.DestinationPortRange)
+			}
+		}
+
+		assert.ElementsMatch(t, test.Config.BastionOutboundRules, actual, fmt.Sprintf("Bastion Subnet doesn't have expected destination ports: %+q", test.Config.BastionOutboundRules))
+	}
+}
+
+func TestJumpboxSecurityRulesCount(t *testing.T) {
+	t.Parallel()
+
+	test := prepareTestTable()
+
+	for iLoop := 1; iLoop <= 2; iLoop++ {
+		rules := azure.GetAllNSGRules(t, fmt.Sprintf("%s-rg-vnet-hub-re%d", test.Prefix, iLoop), fmt.Sprintf("%s-nsg-jumpbox", test.Prefix), test.SubscriptionID)
```

Comment:
Minor future enhancement. The tests should accept a pattern for the different resource names since these come from a config file. Ideally we would read the configuration tfvars. Not a blocker for this time, but just noting it as a potential future enhancement.

Reply:
yeah, I'm gonna add an item in the backlog to remove the config file and "parse the configuration.tfvars" file. so we won't have to generate the name of the resources and resource groups.

if that's OK, can you resolve this comment? otherwise, I'll add a config item